Source/Core/Core/NetPlayServer.cpp

```diff
@@ -2085,6 +2030,93 @@ bool NetPlayServer::PlayerHasControllerMapped(const PlayerId pid) const
          std::any_of(m_wiimote_map.begin(), m_wiimote_map.end(), mapping_matches_player_id);
 }
 
+void NetPlayServer::AssignNewUserAPad(const Client& player)
+{
+  for (PlayerId& mapping : m_pad_map)
+  {
+    if (mapping == Unmapped)
+    {
+      mapping = player.pid;
+      break;
+    }
+  }
+}
+
+PlayerId NetPlayServer::GiveFirstAvailableIDTo(ENetPeer* player)
+{
+  PlayerId pid = 1;
+  for (auto i = m_players.begin(); i != m_players.end(); ++i)
+  {
+    if (i->second.pid == pid)
+    {
+      pid++;
+      i = m_players.begin();
+    }
+  }
+  player->data = new PlayerId(pid);
+  return pid;
+}
+
+void NetPlayServer::SendResponse(MessageID message_id, const Client& player)
+{
+  sf::Packet response;
+  switch (message_id)
+  {
+  case MessageID::PlayerJoin:
+    response << MessageID::PlayerJoin;
+    response << player.pid << player.name << player.revision;
+    break;
+
+  case MessageID::ConnectionSuccessful:
+    response << MessageID::ConnectionSuccessful;
+    response << player.pid;
+    break;
+
+  case MessageID::ChangeGame:
+    response << MessageID::ChangeGame;
+    SendSyncIdentifier(response, m_selected_game_identifier);
+    response << m_selected_game_name;
+    break;
+
+  case MessageID::PadBuffer:
+    response << MessageID::PadBuffer;
+    response << m_target_buffer_size;
+    break;
+
+  case MessageID::HostInputAuthority:
+    response << MessageID::HostInputAuthority;
+    response << m_host_input_authority;
+    break;
+  default:
+    INFO_LOG_FMT(NETPLAY, "Warning! Call to SendResponse() failed to send a packet.");
+    return;
+    break;
+  }
+
+  // no player specified
+  if (player == Client{})
+    SendToClients(response);
+  else
+    Send(player.socket, response);
+}
+
+void NetPlayServer::TellNewPlayerAboutExistingPlayers(const Client& new_player)
```

Comment:
`AssignNewUserAPad` and `GiveFirstAvailableIDTo`  are good extractions. 

But if you simplify along the lines as above, the body of `TellNewPlayerAboutExistingPlayers` one will condense down to just 4 lines, and you would be better off not-extracting.